dll/win32/shell32/wine/control.c

```diff
@@ -967,65 +960,199 @@ static	void	Control_DoLaunch(CPanel* panel, HWND hWnd, LPCWSTR wszCmd)
             sp = 0;
         }
 
-#ifdef __REACTOS__
-        bActivated = (applet->hActCtx != INVALID_HANDLE_VALUE ? ActivateActCtx(applet->hActCtx, &cookie) : FALSE);
+        if (!applet->proc(applet->hWnd, CPL_STARTWPARMSW, sp, (LPARAM)extraPmts))
+            applet->proc(applet->hWnd, CPL_DBLCLK, sp, applet->info[sp].data);
+
+        Control_UnloadApplet(applet);
+    }
 
-        aCPLPath = GlobalFindAtomW(applet->cmd);
-        if (!aCPLPath)
+    HeapFree(GetProcessHeap(), 0, buffer);
+#else
+    LPWSTR	buffer;
+    LPWSTR       ptr;
+    signed 	sp = -1;
+    LPCWSTR	extraPmts = L"";
+    BOOL        quoted = FALSE;
+    CPlApplet *applet;
+    LPCWSTR wszFirstCommaPosition = NULL, wszSecondCommaPosition = NULL;
```

Comment:
Regarding these variable, I guess it's understood it's about their "position" (well, a pointer to them in the string; because otherwise "position" would imply the variable is a numerical offset, which it is not). So you may omit that from the variable names.
Also, "wsz" --> "pch" (pointer to character)